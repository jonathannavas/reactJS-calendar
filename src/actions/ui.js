
import { types } from "../types/types";

// TODO Aqui van las acciones del UI

export const uiOpenModal = () => ({
    type: types.uiOpenModal
});

export const uiCloseModal = () => ({
    type: types.uiCloseModal
});