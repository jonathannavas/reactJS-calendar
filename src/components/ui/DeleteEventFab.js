import React from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { eventStartDelete } from '../../actions/events';

export const DeleteEventFab = () => {

    // const {activeEvent} = useSelector(state => state.calendar);
    const dispatch = useDispatch();

    const handleDelete = () => {
        dispatch( eventStartDelete() );
    }

    return (
        <button className="btn btn-danger fab-danger" onClick={handleDelete}>
            <i className="fas fa-trash"></i>
            <span>Borrar evento</span>
        </button>
    )
}
