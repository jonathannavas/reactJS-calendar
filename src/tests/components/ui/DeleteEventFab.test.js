import React from 'react';
import '@testing-library/jest-dom';
import { mount } from "enzyme";
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'; 
import { DeleteEventFab } from "../../../components/ui/DeleteEventFab";
import { eventStartDelete } from '../../../actions/events';

jest.mock('../../../actions/events', () => ({
    eventStartDelete: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
const store = mockStore(initState);

store.dispatch = jest.fn();
const wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab />
    </Provider>
);

describe('Tests on DeleteEventFab.js', () => {
    test('should make a snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Should call eventStartDelete', () => {
        wrapper.find('.fab-danger').prop('onClick')();
        expect(eventStartDelete).toHaveBeenCalled();
    });
});
