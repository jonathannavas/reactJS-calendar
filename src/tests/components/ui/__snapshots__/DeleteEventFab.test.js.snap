// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on DeleteEventFab.js should make a snapshot 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
    Borrar evento
  </span>
</button>
`;
